pkg/chains/storage/docdb/docdb.go

```diff
@@ -145,13 +128,23 @@ func WatchBackend(ctx context.Context, cfg config.Config, watcherStop chan bool)
 					continue
 				}
 
-				updatedEnv, err := getMongoServerURLFromDir(cfg.Storage.DocDB.MongoServerURLDir)
-				if err != nil {
-					logger.Error(err)
-					backendChan <- nil
+				var updatedEnv string
+				if cfg.Storage.DocDB.MongoServerURLPath != "" {
+					updatedEnv, err = getMongoServerURLFromPath(cfg.Storage.DocDB.MongoServerURLPath)
+					if err != nil {
+						logger.Error(err)
+						backendChan <- nil
+					}
+				} else if cfg.Storage.DocDB.MongoServerURLDir != "" {
+					updatedEnv, err = getMongoServerURLFromDir(cfg.Storage.DocDB.MongoServerURLDir)
+					if err != nil {
+						logger.Error(err)
+						backendChan <- nil
+					}
 				}
+
 				if updatedEnv != os.Getenv("MONGO_SERVER_URL") {
-					logger.Infof("directory %s has been updated, reconfiguring backend...", cfg.Storage.DocDB.MongoServerURLDir)
+					logger.Infof("Mongo server url %s has been updated, reconfiguring backend...", cfg.Storage.DocDB.MongoServerURLPath)
```

Comment:
We don't know at this point if it was updated because of a change in `cfg.Storage.DocDB.MongoServerURLDir` or `cfg.Storage.DocDB.MongoServerURLPath`. Maybe just don't include the path at all in the log message?